### ISSUE TITLE
copilot-cli v1.24.0 causes AmazonECS ServerException when deploying

### ISSUE THREAD

**Issue by jryanmcc** [aws/copilot-cli#4353]
Running in *us-gov-west-1*, this command successfully deployed in copilot-cli v1.23.0:

`TAG=7dcd52058990a783e68906f0dba541fce373c2fc copilot deploy --env test --name web`

Now, with copilot-cli v1.24.0:

```
❯ TAG=7dcd52058990a783e68906f0dba541fce373c2fc copilot deploy --env test --name web
✔ Proposing infrastructure changes for stack [REDACTED]
- Updating the infrastructure for stack [REDACTED]                          [update rollback complete]  [26.6s]
  The following resource(s) failed to update: [Service].                                                         
  - An ECS service to run and maintain your tasks in the environment cluster         [update complete]           [1.8s]
    Resource handler returned message: "Service was unable to process this                                       
     request. (Service: AmazonECS; Status Code: 500; Error Code: ServerExc                                       
    eption; Request ID: 1157666c-74e1-4fe8-ba75-eb43fc9f0ff5; Proxy: null)                                       
    " (RequestToken: 6ca35636-d8cf-0fcc-13eb-b744abc4de41, HandlerErrorCod                                       
    e: GeneralServiceException)                                                                                  
    Deployments                                                                                                   
               Revision  Rollout      Desired  Running  Failed  Pending                                                   
      PRIMARY  11        [completed]  1        1        0       0                                                         
  - An ECS task definition to group your containers and run them on ECS              [delete complete]           [0.0s]
```

It's interesting that with v1.24.0 it doesn't even appear to advance the ECS task revision... it just shows the current version (in this case, 11, is already running).

web/manifest.yml
```
name: web
type: Load Balanced Web Service

http:
  path: '/'
  healthcheck: '/health'

image:
  location: [REDACTED]:${TAG}
  port: 8000

cpu: 256
memory: 1024
platform: linux/x86_64
count:
  spot: 1
exec: true
entrypoint: ["gunicorn", "-b", "0.0.0.0:8000", "--workers", "4", [REDACTED], "--timeout", "120"]

environments:
  test:
    http:
      alias:
      [REDACTED]
    variables:
      [REDACTED]

    secrets:
      [REDACTED]
network:
  vpc:
    placement: 'private'
```

environments/test/manifest.yml
```
name: test
type: Environment

network:
  vpc:
    id: [REDACTED]
    subnets:
      public:
        - id: [REDACTED]
        - id: [REDACTED]
      private:
        - id: [REDACTED]
        - id: [REDACTED]

http:
  public:
    certificates:
      [REDACTED]

observability:
  container_insights: false
```

**Comment by Lou1415926** [comment]
Hello @jryanmcc ! Weird error 🤔 Is this error persistent? 

> "Service was unable to process this                                       
     request. (Service: AmazonECS; Status Code: 500; Error Code: ServerExc                                       
    eption; Request ID: 1157666c-74e1-4fe8-ba75-eb43fc9f0ff5; Proxy: null) 

The error code is `ServerException` - it looks like the server might have a glitch at the time.

Related reference: https://aws.amazon.com/premiumsupport/knowledge-center/ecs-api-common-errors/#ServerException

**Comment by jryanmcc** [comment]
Yes, the ServerException is persistent. I can reliably reproduce it by switching from v1.23.0 to v1.24.0. Once I switch back to v1.23.0 I can deploy.

I didn't see anything in the release notes for 1.24 that looks like it would cause this, but I haven't reviewed all the commits between 1.23 and 1.24 yet. We are running in GovCloud and in the past I've seen some tools make assumptions that don't hold true in other AWS partitions (i.e. GovCloud).

**Comment by Lou1415926** [comment]
@jryanmcc One guess is that starting v1.24  we have this line in the CFN template: https://github.com/aws/copilot-cli/blob/c4bb8ece8e5d9688fb549e9a4500a8f6b29862b5/internal/pkg/template/templates/workloads/partials/cf/service-base-properties.yml#L76 

Maybe CFN makes an API call to ECS with an explicitly turned off ServiceConnect. Since ServiceConnect isn't supported in gov cloud yet (according to [this](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-connect.html)), the server refuses to process an API call with ServiceConnect configuration, even if it is saying "off".

I will have to verify this hypothesis with someone! I will post any update here.

**Comment by Lou1415926** [comment]
@jryanmcc Seems like my guess ⬆️  is correct.

@iamhopaul123 has tagged this issue as a bug. The fix would probably to not render `ServiceConnectConfiguration` at all for govcloud users. 

Sorry for the trouble!! Please stay in `v1.23.0` for now until we have a fix out 🙇🏼

**Comment by iamhopaul123** [comment]
Hello @jryanmcc. The fix has been merged. However, we won’t be able to test this fix since the gov-cloud account is still not available yet. Feel free to pull the latest commit and test if it works when service connect is disabled.

Alternatively if you prefer to wait for validation, it’ll take us a little bit more time to make sure the version merged in mainline is safe but we will write here once we test it.

**Comment by iamhopaul123** [comment]
Hello! The fix is now released in v1.26.0: https://github.com/aws/copilot-cli/releases/tag/v1.26.0

**Comment by jryanmcc** [comment]
Seems to work well, thank you!
